core/include/kernel/pm.h

```diff
@@ -0,0 +1,112 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2018, Linaro Limited
+ */
+
+#ifndef __KERNEL_PM_H
+#define __KERNEL_PM_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+/*
+ * Platform hints on targeted power state. Lower bits
+ * are generic resource bit flags. Higher bits are platform
+ * specific. Registered callbacks may choose to use or ignore
+ * these hints.
+ *
+ * PM_STATE_HINT_CLOCK - When set clock shall be suspended/restored
+ * PM_STATE_HINT_POWER - When set device power shall be suspended/restored
+ * PM_STATE_HINT_IO - When set IO pins shall be suspended/restored
+ * PM_STATE_HINT_CONTEXT - When set the full context be suspended/restored
+ * PM_STATE_PLATFORM_MASK - Bit mask reserved for platform specific hints
+ */
+#define PM_STATE_HINT_CLOCK		BIT(0)
+#define PM_STATE_HINT_POWER		BIT(1)
+#define PM_STATE_HINT_IO		BIT(2)
+#define PM_STATE_HINT_CONTEXT		BIT(3)
+#define PM_STATE_PLATFORM_MASK		GENMASK_32(31, 16)
+
+/*
+ * PM_OP_SUSPEND: platform is suspending to a target low power state
+ * PM_OP_RESUME: platform is resuming from low power state
+ */
+enum pm_op {
+	PM_OP_SUSPEND = 0,
+	PM_OP_RESUME = 1,
+};
+
+/*
+ * Registered callbacks are called the ordering directives
+ * specified by the PM_CB_LEVEL_xxx value.
+ */
+enum pm_cb_level {
```

Comment:
Since this is a pure ordering thing I'd suggest `enum pm_callback_order`, it's fine with `PM_CB_ORDER_` prefix for the values.